include/ofi_mr.h

```diff
@@ -338,26 +350,48 @@ static inline bool ofi_mr_cache_full(struct ofi_mr_cache *cache)
 
 bool ofi_mr_cache_flush(struct ofi_mr_cache *cache, bool flush_lru);
 
-int ofi_mr_cache_search(struct ofi_mr_cache *cache, const struct fi_mr_attr *attr,
+/**
+ * @brief Given an ofi_mr_info (with an iov range, ipc_info)
+ * If the iov range is already registered and validated by the monitor, assign the corresponding
+ * ofi_mr_entry to entry. Otherwise, register a new ofi_mr_entry and
+ * assign it to entry.
+ * The ofi_mr_info.ipc_info is used by ipc monitor to check whether the ipc handle stored in
+ * the queried mr_entry is valid. It will be ignored if the cache monitor is NOT ipc monitor.
+ *
+ * @param[in] cache     The cache the entry belongs to
+ * @param[in] info      Information about the mr entry to search
+ * @param[out] entry    The registered entry corresponding to the
+ *			region described in info.
+ * @returns On success, returns 0. On failure, returns a negative error code.
+ */
+int ofi_mr_cache_search(struct ofi_mr_cache *cache, const struct ofi_mr_info *info,
```

Comment:
Move changes to the existing cache API into its own patch.

Reply:
With the latest design, there is no change in the existing cache API. I just need to make `ofi_mr_cache_search` initialize the `info` to `{0}` inside the function. And I made it a separate commit.